src/vs/workbench/api/common/extHostTreeViews.ts

```diff
@@ -24,6 +24,10 @@ import { IMarkdownString } from 'vs/base/common/htmlContent';
 
 type TreeItemHandle = string;
 
+function isMarkdownString(value: any): value is vscode.MarkdownString {
+	return (value !== undefined) && value.appendCodeblock && value.appendMarkdown && value.appendText && (value.value !== undefined);
+}
+
```

Comment:
lgtm but for future stability you should have `MarkdownString#isMarkdownString`, like [`Range#isRange`](https://github.com/microsoft/vscode/blob/348e2b4c6ef165a03847329e5efc8e999ec30bc1/src/vs/workbench/api/common/extHostTypes.ts#L234). Start with an instance-of check and then some looks-like-checks